src/extensions/billingMigrationHelper.ts

```diff
@@ -41,7 +41,16 @@ function hasRuntime(spec: ExtensionSpec, runtime: string): boolean {
   const specVersion = spec.specVersion || defaultSpecVersion;
   const defaultRuntime = defaultRuntimes[specVersion];
   const resources = spec.resources || [];
-  return resources.some((r) => runtime === (r.properties?.runtime || defaultRuntime));
+  return resources.some((r) => {
+    switch (r.type) {
```

Comment:
Seems like this could use getResourceRuntime to dedupe code a bit?

Reply:
Ah, good catch! Done!